### PR TITLE
Vet center spike

### LINKED ISSUE #22099: Spike solution for Vet Center Main page
## Issue Description
This issue is to generate the skeleton of the Vet Center main page.

---
## Tasks
- [ ] Generate the Vet Center main pages using GraphQL to access Vet Center data from Drupal. 

## Acceptance Criteria
- [ ] Page resembles prototype
- [ ] Page contains content from CMS,
- [ ] Page uses the build and components outlined in the architecture plan #22071
- [ ] Pages can be viewed in Tugboat and is not available in prod

### PR DESCRIPTION
**Pull request by mpelzsherman**
Resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/22099 and https://github.com/department-of-veterans-affairs/va.gov-team/issues/22100

Note: this PR will not impact production because no Vet Center content has been published yet.

### CODE REVIEW

**Review by brangi — APPROVED** [review]

**Review by timwright12 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "jsdoc": "jsdoc",
     "prebuild": "node script/prebuild.js",
     "preview": "node script/preview.js",
+    "preview:build": "yarn build:webpack --env.entry=static-pages && yarn preview",
```
Comment (ncksllvn) [inline package.json:34]: you would think i'd know this answer but i don't...don't we also need to run `yarn build:content` to run preview?

File: src/site/layouts/vet_center.drupal.liquid [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{% include "src/site/includes/preview-edit.drupal.liquid" %}
+{% include "src/site/includes/breadcrumbs.drupal.liquid" with hideHomeBreadcrumb = false %}
+
+<div class="interior" id="content">
+  <main class="va-l-detail-page va-facility-page">
+    <div class="usa-grid usa-grid-full">
+      <div class="usa-width-three-fourths">
+        <article class="usa-content va-l-facility-detail">
+          {% if entityLabel != empty %}
+            <h1>{{ entityLabel }}</h1>
```
Comment (ncksllvn) [inline src/site/layouts/vet_center.drupal.liquid:12]: i'm not familiar with this entity (obviously) but typically there is a `entity.title` property we use for the first one. There is some logic in the header file that might be worth checking out to make sure this works okay with the meta tags. Or just check the page source of this page.`src/site/includes/metatags.drupal.liquid`

Comment (ncksllvn) [reply src/site/layouts/vet_center.drupal.liquid:12]: also i feel like the `h1` should be required here?

File: src/site/layouts/vet_center_locations_list.drupal.liquid [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{% include "src/site/includes/preview-edit.drupal.liquid" %}
+{% include "src/site/includes/breadcrumbs.drupal.liquid" with hideHomeBreadcrumb = false %}
+
+<div class="interior" id="content">
+  <main class="va-l-detail-page va-facility-page">
+    <div class="usa-grid usa-grid-full">
+      <div class="usa-width-three-fourths">
+        <article class="usa-content va-l-facility-detail">
+          {% if entityLabel != empty %}
+            <h1>{{ entityLabel }}</h1>
```
Comment (ncksllvn) [inline src/site/layouts/vet_center_locations_list.drupal.liquid:12]: It seems like the `h1` should be required here as well?

File: src/site/stages/build/drupal/graphql/vetCenter.graphql.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+const vetCenterFragment = `
+ fragment vetCenterFragment on NodeVetCenter {
+        entityId
+        entityUrl {
+          path
+          routed
+        }
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/vetCenter.graphql.js:7]: There is a shared list of properties that we generally drop into GraphQL queries that represent entire pages, `entityElementsForPages.graphql`. Might be worth checking out

File: src/site/stages/build/drupal/graphql/vetCenter.graphql.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        entityLabel
+        fieldIntroText
+        fieldOfficeHours {
+          day
+          starthours
+          endhours
+          comment
+        }
+        # Other locations link:
+        reverseFieldOfficeNode(limit: 500, filter:{conditions: [{field: "type", value: ["vet_center_locations_list"]}]}) {
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/vetCenter.graphql.js:18]: Ohh `reverseFieldOfficeNode`, very fancy

File: src/site/stages/build/drupal/graphql/vetCenter.graphql.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        fieldPrepareForVisit {
+          entity {
+            ... on ParagraphBasicAccordion {
+              fieldHeader
+              fieldRichWysiwyg {
+                processed
+              }
+            }
+          }
+        }
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/vetCenter.graphql.js:37]: We have some shared paragraph fragments here, `src/site/stages/build/drupal/graphql/fragments.graphql.js`. If you find yourself repeating this block, might be worth defining a fragment for `ParagraphBasicAccordion`

`src/site/stages/build/drupal/graphql/page.graphql.js` is an example of then importing the fragments into your layout query

File: src/site/stages/build/drupal/graphql/vetCenter.graphql.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        ... vetCenterFragment
+      }
+    }
+  }
+`;
+
+module.exports = {
+  fragment: vetCenterFragment,
+  GetVetCenters,
+};
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/vetCenter.graphql.js:83]: If there is going to be a lot of instances of these, you may want to implement or write a ticket for implementing pagination. See `src/site/stages/build/drupal/graphql/vaFormPage.graphql.js` for an example

File: src/site/stages/build/drupal/graphql/vetCenterLocations.graphql.js [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+            ... on NodeVetCenterCap {
+              title
+              entityBundle
+              fieldMedia {
+                entity {
+                  ... on MediaImage {
+                    image {
+                      alt
+                      title
+                      derivative(style: _32MEDIUMTHUMBNAIL) {
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/vetCenterLocations.graphql.js:128]: just wondering - how did you land on using `_32MEDIUMTHUMBNAIL` as the value? was that in the GraphQL Explorer?

**Review by ncksllvn — APPROVED** [review]
Looks great!! Just left a few questions/suggestions

File: package.json (hunk 1 above)
Comment (brangi) [reply package.json:34]: @ncksllvn, isn't it `preview:build` command trying to avoid `yarn build:content`  because it's very slow to rebuild again ?

Comment (ncksllvn) [reply package.json:34]: yeah it just seems like a command `preview:build` would do everything you need to get setup for using the preview server

Comment (mpelzsherman) [reply package.json:34]: I've found that we do, in fact, need to run `yarn build:webpack --env.entry=static-pages` to generate the manifest file _even if_ we've already run `yarn build:content`.

File: src/site/layouts/vet_center.drupal.liquid (hunk 2 above)
Comment (mpelzsherman) [reply src/site/layouts/vet_center.drupal.liquid:12]: Thanks @ncksllvn - we'll check with the CMS folks on this. This is just a quick "spike" to get the page started; we have a lot more work to do on this!

File: src/site/stages/build/drupal/graphql/vetCenterLocations.graphql.js (hunk 8 above)
Comment (mpelzsherman) [reply src/site/stages/build/drupal/graphql/vetCenterLocations.graphql.js:128]: It was in GraphQL Explorer, and we were told to use it by the CMS team during an interactive coding session.

**Pull request merged**